Race tests fail in docker build and are ignored

When I run:
```sh
docker build --pull -t k-rail:manual .
```
I get the following output:
```sh
...
go: finding golang.org/x/text v0.3.0
GO111MODULE=on CGO_ENABLED=1 go test -race -cover github.com/cruise-automation/k-rail/cmd github.com/cruise-automation/k-rail/policies github.com/cruise-automation/k-rail/policies/ingress github.com/cruise-automation/k-rail/policies/pod github.com/cruise-automation/k-rail/resource github.com/cruise-automation/k-rail/server
# runtime/cgo
exec: "gcc": executable file not found in $PATH
FAIL	github.com/cruise-automation/k-rail/policies [build failed]
FAIL	github.com/cruise-automation/k-rail/policies/ingress [build failed]
FAIL	github.com/cruise-automation/k-rail/policies/pod [build failed]
FAIL	github.com/cruise-automation/k-rail/resource [build failed]
FAIL	github.com/cruise-automation/k-rail/server [build failed]
make: *** [Makefile:20: test] Error 2
```
But the docker build is not aborted and completes with exit code 0

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,16 +1,15 @@
 # Build stage
 ARG GO_VERSION=1.13
-FROM golang:${GO_VERSION}-alpine AS builder
-RUN mkdir build; apk --no-cache add ca-certificates git make
-WORKDIR /go/src/github.com/cruise-automation/k-rail
+FROM golang:${GO_VERSION}-buster AS builder
+WORKDIR /build
 COPY ./ /build/
-RUN cd /build; make test; make build
+RUN make test
+RUN make build
 
 # Production image build stage
 FROM scratch
 EXPOSE 8443/tcp
 COPY --from=builder /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/ca-certificates.crt
-COPY --from=builder /etc/passwd /etc/passwd
 COPY --from=builder /build/k-rail /k-rail
 USER 65534
 ENTRYPOINT ["/k-rail", "-config", "/config/config.yml"]

--- a/Dockerfile.local
+++ b/Dockerfile.local
@@ -1,11 +1,9 @@
 FROM alpine as builder
 RUN apk add --no-cache ca-certificates
-USER nobody
 
 FROM alpine
 EXPOSE 8443/tcp
 COPY --from=builder /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/ca-certificates.crt
-COPY --from=builder /etc/passwd /etc/passwd
 COPY k-rail /k-rail
-USER nobody
+USER 65534
 ENTRYPOINT ["/k-rail", "-config", "/config/config.yml"]

